rasa_sdk/endpoint.py

```diff
@@ -90,7 +93,26 @@ async def health(_) -> HTTPResponse:
     @app.post("/webhook")
     async def webhook(request: Request) -> HTTPResponse:
         """Webhook to retrieve action calls."""
-        action_call = request.json
+        if request.headers.get("Content-Encoding") == DEFLATED_REQUEST_BODY_HEADER:
+            try:
+                # Decompress the request data using zlib
+                decompressed_data = zlib.decompress(request.body)
+                # Load the JSON data from the decompressed request data
+                action_call = json.loads(decompressed_data)
+            except zlib.error as e:
+                logger.debug(e)
+                body = {"error": "Error while decompressing body of HTTP request"}
+                return response.json(body, status=400)
+
+            try:
+                action_call = json.loads(decompressed_data)
+            except JSONDecodeError as e:
+                logger.debug(e)
+                body = {"error": e.msg}
+                return response.json(body, status=400)
```

Comment:
```suggestion
            try:
                # Decompress the request data using zlib
                decompressed_data = zlib.decompress(request.body)
                # Load the JSON data from the decompressed request data
                action_call = json.loads(decompressed_data)
             except zlib.error as e:
                logger.debug(e)
                body = {"error": "Error while decompressing body of HTTP request"}
                return response.json(body, status=400)
            except JSONDecodeError as e:
                logger.debug(e)
                body = {"error": e.msg}
                return response.json(body, status=400)
```
(indentation seems messed up, sorry)


The last block in your code duplicates some of the work I think. Additionally, you'll need to define `action_call = None` before the `if` at line 96, otherwise there's at least a branch of code where the `action_call` variable is not defined

Reply:
Oh, I missed that try-catch duplication. Thx for spotting that!

I believe we do not need to set action_call = None, cause there is an if-else branch which will cover initialization of action_call.

Reply:
you're right -- must have been confused